cmd/kubelet/app/server.go

```diff
@@ -933,9 +934,22 @@ func buildKubeletClientConfig(ctx context.Context, s *options.KubeletServer, nod
 	}
 
 	kubeClientConfigOverrides(s, clientConfig)
-	closeAllConns, err := updateDialer(clientConfig)
-	if err != nil {
-		return nil, nil, err
+	// Kubelet needs to be able to recover from stale http connections.
+	// HTTP2 has a mechanism to detect broken connections by sending periodical pings.
+	// HTTP1 only can have one persistent connection, and it will close all Idle connections
+	// once the Kubelet heartbet fails. However, since there are many edge cases that we can't
+	// control, users can still opt-in to the previous behavior for closing the connections by
+	// setting the environment variable DISABLE_HTTP2.
+	var closeAllConns func()
+	if s := os.Getenv("DISABLE_HTTP2"); len(s) > 0 {
```

Comment:
Should we add some log like below if http2 is disabled?
https://github.com/kubernetes/kubernetes/blob/b0bc8adbc2178e15872f9ef040355c51c45d04bb/staging/src/k8s.io/apimachinery/pkg/util/net/http.go#L135-L137

Reply:
I thought that will be logged anyway, because kubelet already calls `SetTransportDefaults`, however, now I don't know if this is the same codepath

```
pkg/kubelet/certificate/transport.go:   clientConfig.Transport = utilnet.SetTransportDefaults(&http.Transport{
```

Reply:
let me add the log with a more meaningful message about this behavior

Reply:
done

Reply:
/lgtm